### PR TITLE
CLDR-17248 Measure vote timing; log times to console; refactor to use fetch

### PR DESCRIPTION
**Pull request by btangmu**
-Make cldrVote.CLDR_VOTE_DEBUG true; log times to console

-Refactor to use modern cldrAjax.doFetch instead of legacy cldrAjax.sendXhr

-Reduce very long functions and nested subroutines

-Sanitize error message with new method makeSafe to avoid security warning

-Comments

CLDR-17248

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-apps/js/src/esm/cldrVote.mjs [hunk 1]
```diff
@@ -553,18 +558,24 @@ function oneLessPendingVote() {
...
+  if (time - lastTimeChanged < 3000) {
     if (CLDR_VOTE_DEBUG) {
-      console.log("cldrVote busy: less than 3 seconds elapsed");
+      console.log("cldrVote busy: less than 3 seconds elapsed; time = " + time);
     }
     return true;
   }
   if (CLDR_VOTE_DEBUG) {
-    console.log("cldrVote not busy");
+    console.log("cldrVote not busy; time = " + time);
```
Comment (srl295) [inline tools/cldr-apps/js/src/esm/cldrVote.mjs:578]: a logging framework  ( <https://getpino.io/#/> is one i've used / contributed to  though not in browser , or something homegrown) could collect timings automatically

**Review by srl295 — APPROVED** [review]
LGTM

File: tools/cldr-apps/js/src/esm/cldrVote.mjs [hunk 2]
```diff
@@ -117,138 +116,134 @@ function handleWiredClick(tr, theRow, vHash, newValue, button) {
...
+  button.checked = false;
+  cldrSurvey.hideLoader();
+}
+
+function handleVoteErr(tr, message, button) {
+  tr.className = "tr_err";
+  tr.innerHTML =
+    "<td colspan='4'>" +
+    cldrStatus.stopIcon() +
+    " Could not check value. Try reloading the page.<br>" +
```
[the remark this answers is not in the document]
Comment (btangmu) [reply tools/cldr-apps/js/src/esm/cldrVote.mjs:216]: I actually tried using cldrNotify and got stuck on some problems with cleaning up the input window, row background color, etc., so I decided not to change too much with this PR

I did some testing by temporarily changing the back end to return Auth.noSessionResponse() or Response.Status.NOT_FOUND, as well as inputting values that triggered errors

The tr_err class shows the messages with black text on a deep red background, not ideal, but one has to pick one's battles :-)

File: tools/cldr-apps/js/src/esm/cldrVote.mjs [hunk 3]
```diff
@@ -553,18 +558,24 @@ function oneLessPendingVote() {
 function isBusy() {
   if (pendingVoteCount > 0) {
     if (CLDR_VOTE_DEBUG) {
-      console.log("cldrVote busy: pendingVoteCount = " + pendingVoteCount);
+      console.log(
```
[the remark this answers is not in the document]
Comment (btangmu) [reply tools/cldr-apps/js/src/esm/cldrVote.mjs:561]: that sounds good!

**Pull request merged**